common/network_manager_utils.go

```diff
@@ -0,0 +1,76 @@
+package common
```

Comment:
I know that @inancgumus is not a fan of lib/helpers/utils files and packages, and neither am I if it can be avoided, but I really don't know what to name this one... Any suggestions?

I didn't want to clutter `network_manager.go` even more and these are only relevant to that file, so I'm not against this too much.

Reply:
:-/ Similarly to discussions in #204, I disagree that functions should be defined inline somewhere just because they're used only there. Making them closures inside `onRequestPaused` would make it too long to be readable (probably triggering the function length linter), and the `check*` functions are a lower level of abstraction than `onRequestPaused`, despite being trivial. I do see a point for `failOrContinueRequest` to be there, though, but again, I prefer having shorter, composable and independent functions than everything being defined in a single place.

I'll see if I can move them without the linter complaining. `failOrContinueRequest` if nothing else.

Then what about `newResolver` and `parseTTL`? They're both also used only once, but I definitely don't want those inline anywhere...

Reply:
I understand your concerns, but the `common` is getting bigger and bigger, and its namespace is getting smaller with each identifier (funcs, etc.). If you don't want to put them inside the same function, at least, create a new type and put those functions as methods around that type. Or if you don't want to do that either, please put them in the same file (network_manager.go) or to a file like: `network.go` or something like that but `utils` 🤷 .

Reply:
> I understand your concerns, but the `common` is getting bigger and bigger, and its namespace is getting smaller with each identifier (funcs, etc.).

`common` will get bigger regardless of where we define functions. It's something we'll have to live with until we start splitting it up.

And is namespace size a practical concern? Is there some limit I'm not aware of? I don't see why we should invent a new type just to avoid polluting the namespace. :man_shrugging: If anything, these could be defined on `NetworkManager` itself, but I chose to keep them independent since they really don't need to be methods.

> if you don't want to do that either, please put them in the same file (`network_manager.go`) or to a file like: network.go or something like that but `utils`.

But they're in `network_manager_utils.go`, i.e. utils for the `NetworkManager`... `network_manager.go` is already 500 lines, but sure, I'll just move them there instead.

You really can't even picture the word `utils` :laughing:

Reply:
OK, this is partly resolved in 24b1fce (sorry, couldn't be bothered to split the changes :sweat_smile:).

`failOrContinueRequest` is now gone and is defined inline for the `defer`. But I left the `block*` functions standalone, and moved them to `network_manager.go`, along with `newResolver` and `parseTTL`, so `network_manager_utils.go` is gone too :tada:

Reply:
By the way, I'd bring those funcs closer to the method that uses them instead of putting them at the end of the file.

Reply:
> these could be defined on NetworkManager itself, but I chose to keep them independent since they really don't need to be methods.

Yes, I agree, but if you had done it that way, it'd still be better than `utils` because they are super-related.

Reply:
> I'd bring those funcs closer to the method that uses them instead of putting them at the end of the file.

I prefer keeping unexported functions near the bottom, and exported structs and methods towards the top. This is more legible for users of the module since they shouldn't care about unexported functions as much, so the reading flow is not interrupted, and they can/should use `go-to-definition` in their IDE to jump to any function regardless of its position in the file.

But sure, I moved them in b8eef0e.

Reply:
> I prefer keeping unexported functions near the bottom, and exported structs and methods towards the top.

This is the way, I agree, and I follow the same principle, but this time is different.

> the reading flow is not interrupted

This is exactly what worries me. Even though they are implementation details, I doubt anyone who reads the `NewNetworkManager` function won't want to read them.